cmd/snap/cmd_snap_op.go

```diff
@@ -414,10 +414,15 @@ func (x *cmdInstall) installOne(name string, opts *client.SnapOptions) error {
 	} else {
 		changeID, err = cli.Install(name, opts)
 	}
-	if e, ok := err.(*client.Error); ok && e.Kind == client.ErrorKindSnapAlreadyInstalled {
-		fmt.Fprintf(Stderr, i18n.G("snap %q is already installed, see \"snap refresh --help\"\n"), name)
+	if e, ok := err.(*client.Error); ok {
+		msg, isError := clientErrorToCmdMessage(name, e)
```

Comment:
I struggled to understand your point here, as nothing seemed to fit a "usual pattern", but suddenly it hit me that maybe what you're talking about is just about returning error as the second argument instead of a boolean. If that's clearer, sure, I'll do that!